src/internal/channel.js

```diff
@@ -113,7 +113,7 @@ export function channel(buffer) {
 }
 
 export function eventChannel(subscribe, buffer = buffers.none(), matcher) {
-  if(arguments.length > 1)
+  if(typeof matcher !== 'undefined')
```

Comment:
Got it. I will make the change so that we fix the documented behavior (2 arguments), but IMO it is the function caller/callee contract that `undefined` is the same as not passing an argument in that position.

I think one of the uses of `undefined` is to indicate no argument, which is why it is used to determine whether to use the default value or not. In fact for this particular function I would make the identity function the default matcher, and not perform any checks at all: `eventChannel(subscribe, buffer = buffers.none(), matcher = x => x)`

The mistake of passing in an undefined variable could be made in any argument position, so if I did that for the second argument, I would unintentionally get your default buffer type and that could also lead to very subtle bugs in my application. Without a default for the third argument we would be handling the two arguments inconsistently.